src/NServiceBus.Extensions.Hosting/ServiceCollectionExtensions.cs

```diff
@@ -19,8 +19,9 @@ public static IServiceCollection AddNServiceBus(this IServiceCollection services
         {
             var startableEndpoint = EndpointWithExternallyManagedContainer.Create(configuration, new ServiceCollectionAdapter(services));
             var hostedService = new NServiceBusHostedService(startableEndpoint);
-
-            services.AddSingleton<IMessageSession>(_ => new WebHostCompatibleMessageSession(hostedService));
+            var webHostCompatibleMessageSession = new WebHostCompatibleMessageSession(hostedService);
+            
+            services.AddSingleton<IMessageSession>(webHostCompatibleMessageSession);
```

Comment:
what's the benefit of this approach?

Reply:
None really but I felt it is more aligned with the collection having locals that then get mapped. And for that local the factory approach is not needed